Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/TrisoftCmdlet.cs

```diff
@@ -35,7 +35,7 @@ namespace Trisoft.ISHRemote.Cmdlets
     ///                and: http://technet.microsoft.com/en-us/library/dd819489.aspx
     /// Progress Record: http://community.bartdesmet.net/blogs/bart/archive/2006/11/26/PowerShell-_2D00_-A-cmdlet-that-reports-progress-_2D00_-A-simple-file-downloader-cmdlet.aspx
     /// </summary>
-    public abstract class TrisoftCmdlet : Cmdlet
+    public abstract class TrisoftCmdlet : PSCmdlet
```

Comment:
I understand the reasoning, but not agree. Having the ability to use the parameter set name to drive a switch condition offers great power in code. Otherwise it makes double checking all parameters very expensive. Switching to `PSCmdlet` is more future proof and if necessary a dll can be separated into a separated isolated project. Is anyone using the the binary assembly as a dll? I notice you are worried about in another comment. How valid is the concern?